vizro-core/src/vizro/static/css/layout.css

```diff
@@ -153,6 +164,18 @@ div.dashboard_container .custom-tooltip {
   max-width: 180px;
   overflow-wrap: break-word;
   padding: var(--spacing-01) var(--spacing-02);
-  text-wrap: wrap;
   white-space: pre-wrap;
 }
+
+#left-sidebar {
+  display: flex;
+  flex-direction: column;
+  width: 64px;
+  background-color: var(--surfaces-bg-02);
+  border-right: 1px solid var(--border-subtle-alpha-01);
+  gap: 40px;
+}
+
+#page-container div[hidden] {
+  display: none;
+}
```

Comment:
This is the important CSS fix @antonymilne 

We've been changing this a few times now but I think the current approach is the least ambiguous one now. Initially we had `className=hidden` that applied `display:none` to the relevant containers, but later decided to return `html.Div(id=XXX, hidden=True)` for containers with IDs while returning `None` for containers without IDs. This lead to different behaviour which became apparent in this PR, as I've provided IDs to the 8 main containers and therefore returned `hidden=True` instead of previously returning `None`.

The odd behaviour was that the containers didn't collapse properly + they took space and created gaps even though there was no content. This can e.g. happen if these hidden containers:
- have padding embedded (in which case the padding is visible even though it's hidden)
- are located in an outer flexbox container with flexbox with a default gap (hidden containers created gaps where there shouldn't be)


**As a reminder:**
- `display:none `: Will not appear on page, visible in the DOM, no space allocated to the components
- `visibility:hidden`: Will not appear on page, visible in the DOM, space is allocated to the components even if hidden
- `None`: Has the advantage of both - not visible in the DOM, not allocating space

So in short - this CSS fix basically applies `display:none` to all hidden Divs automatically. We can apply the same rule as before, but now the visual behaviour should be the same regardless whether we return hidden=True divs or None.

Reply:
Makes complete sense, thank you for the full explanation!